src/shared/lib/validation/proposal/sprint-with-us.ts

```diff
@@ -83,34 +85,47 @@ export function validateSWUProposalReferences(raw: any): ArrayValidation<CreateS
   return validateArrayCustom(raw, validateSWUProposalReference, {});
 }
 
-// TODO validate word limit
 export function validateSWUProposalTeamQuestionResponseResponse(raw: string, wordLimit = MAX_TEAM_QUESTION_WORD_LIMIT): Validation<string> {
-  return validateGenericString(raw, 'Response', 1);
+  if (count(raw, 'words', {}) > wordLimit) {
+    return invalid([`The question response must be less than ${wordLimit} words.`]);
+  } else {
+    return validateGenericString(raw, 'Response', 1);
+  }
 }
 
-export function validateSWUProposalTeamQuestionResponseOrder(raw: number): Validation<number> {
-  return validateNumber(raw, 1, MAX_TEAM_QUESTIONS, 'Order');
+export function validateSWUProposalTeamQuestionResponseOrder(raw: number, opportunityTeamQuestions: SWUTeamQuestion[]): Validation<number> {
+  return validateNumber(raw, 1, opportunityTeamQuestions.length, 'Order');
 }
 
-export function validateSWUProposalTeamQuestionResponse(raw: any): Validation<CreateSWUProposalTeamQuestionResponseBody, CreateSWUProposalTeamQuestionResponseValidationErrors> {
-  const validatedResponse = validateSWUProposalTeamQuestionResponseResponse(getString(raw, 'response'));
-  const validatedOrder = validateSWUProposalTeamQuestionResponseOrder(getNumber(raw, 'order'));
-  if (allValid([validatedResponse, validatedOrder])) {
+export function validateSWUProposalTeamQuestionResponse(raw: any, opportunityTeamQuestions: SWUTeamQuestion[]): Validation<CreateSWUProposalTeamQuestionResponseBody, CreateSWUProposalTeamQuestionResponseValidationErrors> {
+  const validatedOrder = validateSWUProposalTeamQuestionResponseOrder(getNumber(raw, 'order'), opportunityTeamQuestions);
+  if (isInvalid(validatedOrder)) {
+    return invalid({
+      order: getInvalidValue(validatedOrder, undefined)
+    });
+  }
+  const wordLimit = opportunityTeamQuestions.find(q => q.order === validatedOrder.value)?.wordLimit || 0;
+  if (!wordLimit) {
+    return invalid({
+      order: ['No matching opportunity question.']
+    });
+  }
```

Comment:
This block will evaluate if the `wordLimit` is zero. Not sure if that is intentional?

Reply:
Yeah, I default it to 0 in the line above if it can't match the order up with the opportunity questions.  To make it clearer though, I should default to null.  Not sure why I picked 0.